ClosedXML/Excel/XLWorkbook_Load.cs

```diff
@@ -400,50 +400,60 @@ private void LoadSpreadsheetDocument(SpreadsheetDocument dSpreadsheet)
                 if (worksheetPart.WorksheetCommentsPart != null)
                 {
                     var root = worksheetPart.WorksheetCommentsPart.Comments;
-                    var authors = root.GetFirstChild<Authors>().ChildElements;
-                    var comments = root.GetFirstChild<CommentList>().ChildElements;
+                    var authors = root.GetFirstChild<Authors>().ChildElements.OfType<Author>().ToList();
+                    var comments = root.GetFirstChild<CommentList>().ChildElements.OfType<Comment>().ToList();
 
                     // **** MAYBE FUTURE SHAPE SIZE SUPPORT
-                    XDocument xdoc = GetCommentVmlFile(worksheetPart);
+                    var shapes = GetCommentShapes(worksheetPart);
 
-                    foreach (Comment c in comments)
+                    for (var i = 0; i < comments.Count; i++)
                     {
+                        var c = comments[i];
+
+                        XElement shape = null;
+                        if (i < shapes.Count)
+                            shape = shapes[i];
+
                         // find cell by reference
                         var cell = ws.Cell(c.Reference);
 
-                        var xlComment = cell.Comment;
+                        var shapeIdString = shape?.Attribute("id")?.Value;
```

Comment:
I don't have any examples of where this attribute is missing, but I'm trying to cover that case.